### PR TITLE
Remove unused/dead code

### PR DESCRIPTION
**Pull request by ghost**
This PR removes unused/dead code

Signed-off-by: Darshna Das <darshna.das98@gmail.com>

#### What type of PR is this?

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/approve
for pkg/scheduler, but I would recommend splitting this PR per sig

File: cluster/images/etcd/migrate/versions.go [hunk 1]
```diff
@@ -187,11 +179,3 @@ func ParseSupportedVersions(list []string) (SupportedVersions, error) {
 	return versions, nil
 }
 
-// MustParseSupportedVersions parses a comma separated list of etcd versions or panics if the parse fails.
-func MustParseSupportedVersions(list []string) SupportedVersions {
```
Comment (dims) [inline cluster/images/etcd/migrate/versions.go:191]: weird, this is being used in a test case, so please don't delete:
https://cs.k8s.io/?q=MustParseSupportedVersions&i=nope&files=&excludeFiles=&repos=kubernetes/kubernetes

File: cluster/images/etcd/migrate/versions.go [hunk 2]
```diff
@@ -124,15 +124,7 @@ func ParseEtcdVersionPair(s string) (*EtcdVersionPair, error) {
 	return &EtcdVersionPair{version, storageVersion}, nil
 }
 
-// MustParseEtcdVersionPair parses a "<version>/<storage-version>" string to an EtcdVersionPair
-// or panics if the parse fails.
-func MustParseEtcdVersionPair(s string) *EtcdVersionPair {
```
Comment (dims) [inline cluster/images/etcd/migrate/versions.go:129]: Same, it's being used https://cs.k8s.io/?q=MustParseEtcdVersionPair&i=nope&files=&excludeFiles=&repos=kubernetes/kubernetes

File: cmd/kube-proxy/app/server.go [hunk 3]
```diff
@@ -91,7 +91,6 @@ const (
 	proxyModeUserspace   = "userspace"
 	proxyModeIPTables    = "iptables"
 	proxyModeIPVS        = "ipvs"
-	proxyModeKernelspace = "kernelspace"
```
Comment (dims) [inline cmd/kube-proxy/app/server.go:94]: 👍🏾

File: pkg/controller/replicaset/replica_set_test.go [hunk 4]
```diff
@@ -504,8 +504,7 @@ func TestPodControllerLookup(t *testing.T) {
 	}
 }
 
-// byName sorts pods by their names.
-type byName []*v1.Pod
```
Comment (dims) [inline pkg/controller/replicaset/replica_set_test.go:508]: 👍🏾

File: pkg/controller/volume/attachdetach/reconciler/reconciler_test.go [hunk 5]
```diff
@@ -39,7 +39,6 @@ const (
 	reconcilerLoopPeriod      time.Duration = 0 * time.Millisecond
 	syncLoopPeriod            time.Duration = 100 * time.Minute
 	maxWaitForUnmountDuration time.Duration = 50 * time.Millisecond
-	resyncPeriod              time.Duration = 5 * time.Minute
```
Comment (dims) [inline pkg/controller/volume/attachdetach/reconciler/reconciler_test.go:42]: 👍🏾

File: pkg/kubelet/apis/config/helpers_test.go [hunk 6]
```diff
@@ -82,31 +82,20 @@ func allPrimitiveFieldPaths(t *testing.T, tp reflect.Type, path *field.Path) set
 //lint:file-ignore U1000 Ignore dummy types, used by tests.
 
 // dummy helper types
-type foo struct {
```
Comment (dims) [inline pkg/kubelet/apis/config/helpers_test.go:85]: 👍🏾

File: pkg/kubelet/certificate/bootstrap/bootstrap_test.go [hunk 7]
```diff
@@ -141,7 +141,6 @@ func TestRequestNodeCertificate(t *testing.T) {
 type failureType int
 
 const (
-	noError failureType = iota
```
Comment (dims) [inline pkg/kubelet/certificate/bootstrap/bootstrap_test.go:144]: 👍🏾

File: pkg/kubelet/dockershim/helpers_linux.go [hunk 8]
```diff
@@ -156,5 +156,4 @@ func getNetworkNamespace(c *dockertypes.ContainerJSON) (string, error) {
 }
 
 // applyExperimentalCreateConfig applys experimental configures from sandbox annotations.
-func applyExperimentalCreateConfig(createConfig *dockertypes.ContainerCreateConfig, annotations map[string]string) {
```
Comment (dims) [inline pkg/kubelet/dockershim/helpers_linux.go:159]: 👍🏾

File: pkg/scheduler/framework/plugins/noderesources/requested_to_capacity_ratio.go [hunk 9]
```diff
@@ -31,7 +31,6 @@ import (
 const (
 	// RequestedToCapacityRatioName is the name of this plugin.
 	RequestedToCapacityRatioName = "RequestedToCapacityRatio"
-	minUtilization               = 0
```
Comment (dims) [inline pkg/scheduler/framework/plugins/noderesources/requested_to_capacity_ratio.go:34]: 👍🏾

File: pkg/util/coverage/fake_test_deps.go [hunk 10]
```diff
@@ -24,7 +24,6 @@ import (
 // no tests are actually run. It does need a concrete implementation of at least ImportPath,
 // which is called unconditionally when running tests.
 //lint:ignore U1000 see comment above, we know it's unused normally.
-type fakeTestDeps struct{}
```
Comment (dims) [inline pkg/util/coverage/fake_test_deps.go:27]: 👍🏾

File: pkg/util/iptables/iptables.go [hunk 11]
```diff
@@ -631,7 +631,6 @@ const (
 	opFlushChain  operation = "-F"
 	opDeleteChain operation = "-X"
 	opListChain   operation = "-S"
-	opAppendRule  operation = "-A"
```
Comment (dims) [inline pkg/util/iptables/iptables.go:634]: 👍🏾

File: pkg/volume/glusterfs/glusterfs.go [hunk 12]
```diff
@@ -77,7 +77,6 @@ const (
 	volPrefix                      = "vol_"
 	dynamicEpSvcPrefix             = "glusterfs-dynamic"
 	replicaCount                   = 3
-	durabilityType                 = "replicate"
```
Comment (dims) [inline pkg/volume/glusterfs/glusterfs.go:80]: 👍🏾

File: pkg/volume/util/volumepathhandler/volume_path_handler.go [hunk 13]
```diff
@@ -31,7 +31,6 @@ import (
 
 const (
 	losetupPath           = "losetup"
-	statPath              = "stat"
```
Comment (dims) [inline pkg/volume/util/volumepathhandler/volume_path_handler.go:34]: 👍🏾

File: pkg/volume/vsphere_volume/vsphere_volume_util.go [hunk 14]
```diff
@@ -36,7 +36,6 @@ import (
 )
 
 const (
-	maxRetries         = 10
```
Comment (dims) [inline pkg/volume/vsphere_volume/vsphere_volume_util.go:39]: 👍🏾

File: plugin/pkg/admission/imagepolicy/certs_test.go [hunk 15]
```diff
@@ -21,33 +21,7 @@ limitations under the License.
 
 package imagepolicy
 
-var caKey = []byte(`-----BEGIN RSA PRIVATE KEY-----
```
Comment (dims) [inline plugin/pkg/admission/imagepolicy/certs_test.go:24]: 👍🏾

File: test/e2e/apimachinery/flowcontrol.go [hunk 16]
```diff
@@ -108,13 +108,8 @@ var _ = SIGDescribe("API priority and fairness", func() {
 		type client struct {
 			username                    string
 			qps                         float64
-			priorityLevelName           string  //lint:ignore U1000 field is actually used
```
Comment (dims) [inline test/e2e/apimachinery/flowcontrol.go:111]: please don't delete this!

Comment (dims) [reply test/e2e/apimachinery/flowcontrol.go:111]: and the ones below.

File: test/e2e/kubectl/kubectl.go [hunk 17]
```diff
@@ -101,7 +101,6 @@ const (
 	httpdDeployment1Filename  = "httpd-deployment1.yaml.in"
 	httpdDeployment2Filename  = "httpd-deployment2.yaml.in"
 	httpdDeployment3Filename  = "httpd-deployment3.yaml.in"
-	httpdRCFilename           = "httpd-rc.yaml.in"
```
Comment (dims) [inline test/e2e/kubectl/kubectl.go:104]: 👍🏾

File: test/e2e/kubectl/kubectl.go [hunk 18]
```diff
@@ -114,9 +113,6 @@ var (
 
 var (
 	proxyRegexp = regexp.MustCompile("Starting to serve on 127.0.0.1:([0-9]+)")
-
-	cronJobGroupVersionResourceAlpha = schema.GroupVersionResource{Group: "batch", Version: "v2alpha1", Resource: "cronjobs"}
```
Comment (dims) [inline test/e2e/kubectl/kubectl.go:118]: 👍🏾

File: test/e2e/network/kube_proxy.go [hunk 19]
```diff
@@ -42,7 +42,6 @@ var kubeProxyE2eImage = imageutils.GetE2EImage(imageutils.Agnhost)
 
 var _ = SIGDescribe("KubeProxy", func() {
 	const (
-		testDaemonHTTPPort    = 11301
```
Comment (dims) [inline test/e2e/network/kube_proxy.go:45]: 👍🏾

File: test/e2e/storage/csi_mock_volume.go [hunk 20]
```diff
@@ -1631,7 +1631,6 @@ func podRunning(ctx context.Context, c clientset.Interface, podName, namespace s
 }
 
 const (
-	podStartTimeout                = 5 * time.Minute
```
Comment (dims) [inline test/e2e/storage/csi_mock_volume.go:1634]: 👍🏾

File: test/e2e/upgrades/apps/replicasets.go [hunk 21]
```diff
@@ -34,8 +34,6 @@ import (
 )
 
 const (
-	interval = 10 * time.Second
```
Comment (dims) [inline test/e2e/upgrades/apps/replicasets.go:37]: 👍🏾

File: test/images/agnhost/net/nat/closewait.go [hunk 22]
```diff
@@ -39,7 +39,6 @@ import (
 // leakedConnection is a global variable that should leak the active
 // connection assigned here.
 //lint:ignore U1000 intentional unused variable
-var leakedConnection *net.TCPConn
```
Comment (dims) [inline test/images/agnhost/net/nat/closewait.go:42]: please don't delete this.

File: cluster/images/etcd/migrate/versions.go (hunk 2 above)
Comment (dims) [reply cluster/images/etcd/migrate/versions.go:129]: please see https://golangci-lint.run/usage/false-positives/ on how to add `//nolint` directives to tell linter to ignore this

**Comment by Darshnadas** [comment]
Hello, @dims I have updated the suggested changes.

**Comment by dims** [comment]
@Darshnadas indicated that someone else can pick this up. thanks for taking a look at this and giving someone else a start!

@supriya-premkumar can you please pick this up?

**Comment by supriya-premkumar** [comment]
> @supriya-premkumar can you please pick this up?
I would love to take this one up. 🙏🏽

File: pkg/controller/replicaset/replica_set_test.go [hunk 23]
```diff
@@ -504,8 +504,7 @@ func TestPodControllerLookup(t *testing.T) {
...
+// byName sorts pods by their names
```
Comment (SergeyKanzhelev) [inline pkg/controller/replicaset/replica_set_test.go:507]: remove comment as well

File: pkg/kubelet/dockershim/helpers_linux.go [hunk 24]
```diff
@@ -154,3 +154,6 @@ func getNetworkNamespace(c *dockertypes.ContainerJSON) (string, error) {
 	}
 	return fmt.Sprintf(dockerNetNSFmt, c.State.Pid), nil
 }
+
+// applyExperimentalCreateConfig applys experimental configures from sandbox annotations.
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/dockershim/helpers_linux.go:158]: is it a new code?

**Comment by dims** [comment]
/close

in favor of https://github.com/kubernetes/kubernetes/pull/100801

**Pull request closed without merging**